internal/workflow/preprocessing.go

```diff
@@ -112,6 +118,64 @@ func (w *PreprocessingWorkflow) Execute(
 
 	localPath := filepath.Join(w.sharedPath, filepath.Clean(params.RelativePath))
 
+	if w.checkDuplicates {
+		// Calculate SIP checksum.
+		ev := result.newEvent(ctx, "Calculate SIP checksum")
+		var checksumSIP activities.ChecksumSIPResult
+		e = temporalsdk_workflow.ExecuteActivity(
+			withFilesysActOpts(ctx),
+			activities.ChecksumSIPName,
+			&activities.ChecksumSIPParams{Path: localPath},
+		).Get(ctx, &checksumSIP)
+		if e != nil {
+			result.systemError(ctx, e, ev, "Calculate SIP checksum failed")
+			return result, nil
+		}
+		ev.Succeed(ctx, "Calculate SIP checksum passed: %s", checksumSIP.Value)
+
+		// Check for duplicate SIP.
+		ev = result.newEvent(ctx, "Check duplicate SIP")
+		var checkDuplicate localact.CheckDuplicateResult
+		e = temporalsdk_workflow.ExecuteLocalActivity(
+			withLocalActOpts(ctx),
+			localact.CheckDuplicate,
+			w.ps,
+			&localact.CheckDuplicateParams{
+				Name:     filepath.Base(localPath),
+				Checksum: checksumSIP.Value,
+			},
+		).Get(ctx, &checkDuplicate)
+		if e != nil {
+			result.systemError(ctx, e, ev, "Check duplicate SIP failed")
+			return result, nil
+		}
+		if checkDuplicate.Duplicated {
+			result.validationError(
+				ctx,
+				ev,
+				"Check duplicate SIP failed",
```

Comment:
Suggested: "SIP is a duplicate". I know this doesn't match the format of the other validation messages, but I think it's important to distinguish between the "check duplicate" activity failing, and rejecting the SIP because it is a duplicate.